elements/pfe-primary-detail/src/pfe-primary-detail.js

```diff
@@ -143,26 +208,66 @@ class PfePrimaryDetail extends PFElement {
    * @param {integer} index The index of the item in the details-nav slot
    */
   _initDetailsNav(detailNavElement, index) {
+    // @todo Drop this quick exit, it's causing issues in components that are copied into a shadow root
     // Don't re-init anything that's been initialized already
-    if (detailNavElement.hasAttribute("role") && detailNavElement.dataset.index && detailNavElement.id) {
+    if (detailNavElement.dataset.index && detailNavElement.id) {
       // Make sure the data-index attribute is up to date in case order has changed
       detailNavElement.dataset.index = index;
       return;
     }
 
-    // Set data-index attribute
-    detailNavElement.dataset.index = index;
+    const createToggleButton = detailNavElement.tagName !== "BUTTON";
+    let toggle = null;
+
+    if (createToggleButton) {
+      // @todo const?
+      let attr = detailNavElement.attributes;
+      toggle = document.createElement("button");
+
+      toggle.innerHTML = detailNavElement.innerHTML;
+      toggle.setAttribute("role", "tab");
+
+      // list of attributes that we DO NOT want to pass to our shadow DOM.
+      const denyListAttributes = ["style", "role"];
+
+      // Copy over attributes from original element that aren't in denyList
+      [...attr].forEach((detailNavElement) => {
+        if (!denyListAttributes.includes(detailNavElement.name)) {
+          toggle.setAttribute(detailNavElement.name, detailNavElement.value);
+        }
+      });
+
+      // Keeping track of tagName which is used in mobile layout to maintain heading order
+      toggle.dataset.wasTag = detailNavElement.tagName;
+
+      // If the detailNavElement does not have a ID, set a unique ID
+      if (!detailNavElement.id) {
+        toggle.setAttribute("id", `pfe-detail-toggle-${Math.random().toString(36).substr(2, 9)}`);
+      }
+    } else {
+      toggle = detailNavElement;
+    }
 
     // If the detailNavElement does not have a ID, set a unique ID
     if (!detailNavElement.id) {
       detailNavElement.setAttribute("id", `pfe-detail-toggle-${Math.random().toString(36).substr(2, 9)}`);
     }
 
-    detailNavElement.setAttribute("role", "tab");
-    detailNavElement.setAttribute("aria-selected", "false");
+    toggle.addEventListener("click", this._handleHideShow);
```

Comment:
This event listener is never removed.

Reply:
Is too! https://github.com/patternfly/patternfly-elements/pull/1582/files#diff-52d62cabced81812fca6be7e6d8b658dda8025d81018b8ca120bdd52b261ca92L133-L137

Reply:
```js
    if (this._slots.detailsNav) {
      for (let index = 0; index < this._slots.detailsNav.length; index++) {
        this._slots.detailsNav[index].removeEventListener("click", this._handleHideShow);
      }
    }
```